cli/checkpoint.go

```diff
@@ -0,0 +1,92 @@
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/big"
+	"strconv"
+
+	eigenpodproofs "github.com/Layr-Labs/eigenpod-proofs-generation"
+	"github.com/Layr-Labs/eigenpod-proofs-generation/cli/onchain"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/fatih/color"
+)
+
+func RunCheckpointProof(ctx context.Context, eigenpodAddress string, eth *ethclient.Client, chainId *big.Int, beaconClient BeaconClient, out, owner *string) {
+	lastCheckpoint := lastCheckpointedForEigenpod(eigenpodAddress, eth)
+	if lastCheckpoint == 0 {
+		if owner != nil {
+			err := startCheckpoint(ctx, eigenpodAddress, *owner, chainId, eth)
+			PanicOnError("failed to start checkpoint", err)
+		} else {
+			PanicOnError("no checkpoint active", errors.New("no checkpoint"))
+		}
+	}
+
+	blockRoot, err := getCurrentCheckpointBlockRoot(eigenpodAddress, eth)
+	PanicOnError("failed to fetch last checkpoint.", err)
+
+	if blockRoot != nil {
+		rootBytes := *blockRoot
+		if AllZero(rootBytes[:]) {
+			PanicOnError("No checkpoint active. Are you sure you started a checkpoint?", fmt.Errorf("no checkpoint"))
+		}
+	}
+
+	headerBlock := "0x" + hex.EncodeToString((*blockRoot)[:])
```

Comment:
`hexutil.Encode` has you covered here since it looks like you're already using go-ethereum: https://github.com/ethereum/go-ethereum/blob/master/common/hexutil/hexutil.go#L84

Reply:
literally the same code lol